state.go

```diff
@@ -1023,7 +1023,14 @@ func (s *State) OnInterface(se *Session, i interface{}) (err error) {
 		}
 	case *GuildEmojisUpdate:
 		if s.TrackEmojis {
-			err = s.EmojisAdd(t.GuildID, t.Emojis)
+			var guild *Guild
+			guild, err = s.Guild(t.GuildID)
+			if err != nil {
+				return err
+			}
+			s.Lock()
```

Comment:
I would personally add a line here, to separate the fetching section, but that's just a style change.
```suggestion
			}
			
			s.Lock()
```